f5_openstack_agent/lbaasv2/drivers/bigip/disconnected_service.py

```diff
@@ -51,43 +76,74 @@ def scan(self):
             loadbalancer_ids = d.plugin_rpc.get_all_loadbalancers()
             for lb in loadbalancer_ids:
                 id = lb['lb_id']
+                if (id in self.timer and self.timer[id]['expired']):
+                    # LB is in error state, no need to check further
+                    continue
                 service = d.plugin_rpc.get_service_by_loadbalancer_id(id)
-                listeners = service.get('listeners', None)
-                service.pop('listeners', None)
-                provisioning_status = \
-                    service['loadbalancer']['provisioning_status']
+
+                if (plugin_const.ACTIVE !=
+                        service['loadbalancer']['provisioning_status']):
```

Comment:
In case that provisioning status ever changes. I'd check it with service['loadbalancer']['provisioning_status'].lower() or .upper(), in case the BIG-IP ever changes this across versions.